store.go

```diff
@@ -240,3 +242,81 @@ func (s *SafeConfigStore) Size() int {
 	defer s.mu.Unlock()
 	return s.c.Size()
 }
+
+// LoggableConfigStore implements the ConfigStore interface and provides a store
+// a field for reference to a logging mechanism to log on reads/writes
+type LoggableConfigStore struct {
+	c   ConfigStore
+	log Logging
+}
+
+// NewLoggableConfigStore takes a Logging and returns a new ConfigStore
+// with said interface as the logging mechanism used for read and writes.
+func NewLoggableConfigStoreWith(l Logging) ConfigStore {
+	return &LoggableConfigStore{
+		c: NewDefaultConfigStore(),
+		log: l,
+	}
+}
+
+// NewLoggableConfigStore returns a ConfigStore with a default logging mechanism
+// set to write to os.Stdout.
+func NewLoggableConfigStore() ConfigStore {
+	l := log.New(os.Stdout, "", 0)
+	return NewLoggableConfigStoreWith(l)
+}
+
+// RegisterResolver registers a custom config resolver for the specified
+// ConfigLevel.
+//
+// Additionally, if the provided level is not already in the current collection
+// of active config levels, it will be added automatically
+func (l *LoggableConfigStore) RegisterResolver(level ConfigLevel, r Resolver) {
+	l.c.RegisterResolver(level, r)
+}
+
+// SetLevel is a generic key/value setter method. It sets the provided k/v at
+// the specified level inside the map, conditionally creating a new ConfigMap if
+// one didn't previously exist.
+func (l *LoggableConfigStore) SetLevel(level ConfigLevel, key string, value interface{}) {
+	l.c.SetLevel(level, key, value)
+	l.log.Print(formatLogLine("SET", level, key, value))
```

Comment:
so, I actually really like the look of this signature but I don't really like the extra call to `formatLogLine` since, I think, that will make it difficult for callers to implement a logger that integrates well with their own log format...

What would your thoughts be on updating the logging interface to be something like this:

```go
type Logging interface {
	LogWrite(level ConfigLevel, key string, val interface{})
	LogRead(key string, val interface{})
}
```
This would put the responsibility of log formatting on the implementer, which is probably more work, but would allow for the log format to integrate with custom logging utilities. For example a logrus `Logging` implementation would look something like 

```go
type LogrusLogger struct {
	log *logrus.Entry
}

func NewLogrusLogger(log *logrus.Entry) Logging {
	return &LogrusLogger{log: log}
}

func (l *LogrusLogger) LogWrite(level ConfigLevel, key string, val interface{}) {
	l.log.WithFields(logrus.Fields{
		"level": level.String(),
		"key": key,
		"val": val,
	}).Info("writing config value")
}

func (l *LogrusLogger) LogRead(key string, val interface{}) {
	l.log.WithFields(logrus.Fields{
		"key": key,
		"val": val,
	}).Info("read config value")
}
```

which can be used like 

```go
func main() {
	logger := logrus.NewLogger()
	// do whatever...
	ven := NewLoggableWith(NewLogrusLogger(logger))
	// keep on doing whatever...
}
```

Then in the `LoggableConfigStore` we can call `LogWrite` and `LogRead` in their respective methods instead of `Log`.